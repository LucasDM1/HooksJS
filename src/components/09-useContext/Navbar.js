import React from 'react'
import { Link } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">

                <h1 className="navbar-brand" href="#">useContext</h1>

                
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                      <Link to="./" className="nav-item nav-link">
                          Home
                      </Link>

                      <Link to="./about" className="nav-item nav-link">
                          About
                      </Link>
                      <Link to="./login" className="nav-item nav-link">
                          login
                      </Link>
                    </div>
                </div>
            </div>
        </nav>
    )
}
