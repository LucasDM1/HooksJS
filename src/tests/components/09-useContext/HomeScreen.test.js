import { mount, shallow } from 'enzyme'
import React from 'react'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en homeScreen', () => {

    const user = {
        name: 'David',
        email: 'lucas19651@hotmail.com',
    }
    
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})

