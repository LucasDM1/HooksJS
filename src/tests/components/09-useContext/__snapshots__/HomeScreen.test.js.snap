// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en homeScreen Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "David",
   "email": "lucas19651@hotmail.com"
}
  </pre>
</div>
`;
